### PR TITLE
Draft: frontend updates

### PR DESCRIPTION
**Pull request by Stefaanna**
- retrieve options from NFT contract
- refactor assets fetching
- call methods for contract
- add frontend input validations
- toast provider for events

### CODE REVIEW

File: src/utils/api.ts [hunk 1]
```diff
@@ -123,8 +126,87 @@ export const fetchNFTDetails = async (
...
+    return true;
+}
+
+export async function loadContractOptions(contract: NFTOpt, setOptionsCallback: (options: Option[]) => void) {
+    let options: Option[] = [];
+    try {
+        const optionsLength = await (await contract.optionID()).toNumber();
+        for (let optionId = 0; optionId <= optionsLength; ++optionId) {
+            const contractOption = await contract.options(optionId);
+            console.log({contractOption});
```
Comment (Stefaanna) [inline src/utils/api.ts:190]:
```suggestion
```

File: src/pages/create-option.tsx [hunk 2]
```diff
@@ -107,20 +111,26 @@ function CreateOption() {
...
+        try {
+            await nftOpt.publishOptionRequest(
+                formState.asset.address,
+                formState.asset.tokenId,
+                ethers.utils.parseEther(`${parseFloat(formState.strikePrice)}`),
+                parseInt(formState.interval) * 24 * 3600, // days in seconds
+                formState.flavor,
+                txOptions
+            );
+            toast.success("Option published successfully", {duration: 4000});
```
Comment (Stefaanna) [inline src/pages/create-option.tsx:129]: make this an util

File: src/utils/api.ts [hunk 3]
```diff
@@ -123,8 +126,86 @@ export const fetchNFTDetails = async (
...
+    return true;
+}
+
+export async function loadContractOptions(contract: NFTOpt, setOptionsCallback: (options: Option[]) => void) {
+    let options: Option[] = [];
+    try {
+        const optionsLength = await (await contract.optionID()).toNumber();
+        for (let optionId = 0; optionId <= optionsLength; ++optionId) {
+            const contractOption = await contract.options(optionId);
+            const option: Option = {
```
Comment (Stefaanna) [inline src/utils/api.ts:190]: make this an util

**Review by grasshopper47 — APPROVED** [review]
LGTM

**Pull request merged**